### ISSUE TITLE
Analyze nullability variance of type and attribute together

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#42470]
Currently, we analyze the nullability variance of types first, then analyze the variance of types+attributes. This results in a spurious warning in cases like the following (reported by @stephentoub )

Relates to https://github.com/dotnet/roslyn/pull/41336 (report bad loosening/tightening in OHI)

```cs
#nullable enable
using System.Diagnostics.CodeAnalysis;

class Program
{
    static void Main() { }
}

interface IFoo<T>
{
    bool TryRead([MaybeNullWhen(false)] out T item);
}

 
class Foo : IFoo<int[]>
{
    public bool TryRead([NotNullWhen(true)] out int[]? item) // warning CS8614: Nullability of reference types in type of parameter 'item' of 'bool Foo.TryRead(out int[]? item)' doesn't match implicitly implemented member 'bool IFoo<int[]>.TryRead(out int[] item)'.
    {
        item = default;
        return false;
    }
}
```

**Comment by stephentoub** [comment]
Another case of this I hit:
```C#
using System.Diagnostics.CodeAnalysis;

class Program
{
    static void Main() { }
}

interface IFoo<T>
{
    [return: MaybeNull]
    T Get();
}

class Foo : IFoo<object>
{
    public object? Get() => null; // warning CS8613: Nullability of reference types in return type of 'object? Foo.Get()' doesn't match implicitly implemented member 'object IFoo<object>.Get()'.
}
```

**Comment by jcouv** [comment]
Fix in progress. Hoping to get it into 16.6p2.

**Comment by jcouv** [comment]
Fix is in for 16.6p3
